client/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
```

Comment:
We should use a cargo workspace rather than a nested crate. Otherwise publishing, lockfiles and caching etc will be more complicated.